core-bundle/src/Resources/contao/drivers/DC_Folder.php

```diff
@@ -2127,6 +2120,36 @@ public function source()
 </form>' . "\n\n" . $codeEditor;
 	}
 
+	private function purgeCache(File $file): void
+	{
+		if (\in_array($file->extension, array('css', 'scss', 'less', 'js')))
+		{
+			$this->import(Automator::class, 'Automator');
+			$this->Automator->purgeScriptCache();
+
+			return;
+		}
+
+		if ('twig' === $file->extension)
+		{
+			$container = System::getContainer();
+
+			$twigCacheDir = Path::makeRelative(
+				$container->get('twig')->getCache(),
+				$container->getParameter('kernel.project_dir')
+			);
+
+			// Always purge the prod cache
+			if ('prod' !== ($env = $container->get('kernel')->getEnvironment()))
+			{
+				$twigCacheDir = str_replace("/$env/", '/prod/', $twigCacheDir);
```

Comment:
Does this work reliably?

Reply:
I think so. I didn't find a better way without hardcoding the directory. 

The only way I'm seeing this could fail is if someone configured the cache dir to contain the path `/prod/` itself which imo is unlikely enough.

Reply:
The only “always working” method would be to execute a command (with the correct environment) that clears the twig cache?

Reply:
We should do it the same way as we do in the Install Tool for example.

Reply:
The only “always working” method would be to execute a command (with the correct environment) that clears the twig cache?

I guess so. But bit of an overkill, isn't it? On CLI people would probably just delete the folder directly (compare https://github.com/twigphp/Twig/issues/1879). 

> We should do it the same way as we do in the Install Tool for example.

Do you mean purging the cache? The install removes all files from `kernel.cache_dir'` there but we always need the prod environment here.

Reply:
We could replace `$kernel_cache_dir/$env/` with `$kernel_cache_dir/$prod/` though, ... :thinking:

Reply:
Why do we need to enforce the prod env?  
IMO editing twig files via the backend in dev mode and expecting the prod cache to be cleared seems weird.

Reply:
As discussed on slack: 69372bc475a710ae9deebc50247dd31ac5b8b4af

This commit drops the logic + always clears the cache of the current environment. If someone switches from debug mode the (app) cache then needs to be cleared manually. On the other hand, this allows editing templates in dev without affecting the cache.